pandas/core/frame.py

```diff
@@ -1964,6 +1965,33 @@ def to_feather(self, path):
 
         to_feather(self, path)
 
+    def to_markdown(self, **kwargs):
```

Comment:
I think my preferred signature here is something like

```
def to_markdown(self, buf=None, mode=None, **kwargs):
    ...
```

In the future, we may have an `engine` keyword for other formatting libraries like tabulate, so we should probably avoid picking too many keywords (like tblfmt, headers, etc.).  

`buf` will have the same semantics as `to_string()` / `to_csv()`. We print to stdout by default but you can also pass your own buffer or filename as a string.

I suspect people will want an option for `index=True/False`. Not sure if we'll want to support that yet.

Reply:
Sure, I've tried updating this as requested.

I haven't added a `pandas/io/formats/markdown.py` file - would that be necessary?

Reply:
Probably not necessary yet.